### ISSUE TITLE
Nested Layout

### ISSUE THREAD

**Issue by mickael-pezzoni** [openopus/ng-pane-manager2#2]
Hello,

I really like your module and I would like to know if it will be possible in a future version to be able to stack windows and move them around ?

Thank you .

**Comment by rookie1024** [comment]
Do you have an example of the behavior you’re describing? If you want the ability to drag and drop groups of horizontally/vertically split windows, I could probably tweak the layout code to allow it.

**Comment by mickael-pezzoni** [comment]
I wish I could have a parent panel that could hold another panel inside and move this window without it being destroyed .
Here is an example:
![Capture d’écran de 2021-01-30 00-41-52](https://user-images.githubusercontent.com/37872697/106338829-152f4280-6295-11eb-8b8f-20ee31cd0961.png)

Do you think this is possible?

**Comment by rookie1024** [comment]
I believe it should be.  If I recall correctly, rendering a header for a pane will make it draggable.  I'll try to fork a mockup demo when I have time.

Is there a use case you have in mind for this feature?  I'll still have to figure out when the layout engine should render draggable headers for split panes.

**Comment by mickael-pezzoni** [comment]
In my case, I would like the user to be able to create his own groups of personalised panels by giving them a title... in which he can group together panels belonging to the same category.    

Thank you for your time

**Comment by rookie1024** [comment]
Okay.  Making the title customizable should be possible, although allowing controls to be placed into title bars so that the user can do so is on the list of features I had been considering, so I’ll take another look at that.  As for allowing the users to create such splits, I can modify the drag’n’drop code to make it possible (e.g. by pressing a modifier key) to join two leaf panels into a draggable split.

**Comment by mickael-pezzoni** [comment]
This would be perfect! and will be it to slide any panel inside once created like any standard panel.
When do you think this could be implemented?

Thanks for everything!

**Comment by rookie1024** [comment]
> And will it be possible to slide any panel inside once created like any standard panel.
Yeah, I should be able to reuse the existing split layout panel, which comes with drag-and-drop out of the box.

> When do you think this could be implemented?
Hard to say, I have a lot of deadlines currently.  Getting the extra headers rendered and updating drag-and-drop behavior shouldn't take too long, but adding support for customizing the header title will probably take longer.

**Comment by rookie1024** [comment]
@mickael-pezzoni Apologies for the radio silence, had some other deadlines to manage.

The easiest way to integrate this feature into the library is by adding a "group pane," which can hold a non-leaf node and render it alongside a custom header.  This should allow for named groups without breaking too much of the rendering or layout tree code.

I've assessed the codebase and drawn up a checklist of changes necessary to implement group panes:
- [x] Modify drag-and-drop code to allow creating groups
- [ ] Modify pane layout structure to allow storing groups
- [ ] Add customizable headers to leaves and groups
- [ ] Handle registration of pane header controls, possibly using a new directive and/or service
- [ ] Handle rendering and instancing custom pane headers in the layout factory
- [ ] Add configuration to the root pane manager to allow or disallow the creation of groups

I'll try to keep https://go.ryan-s.net/apm/ up-to-date with the features I'm implementing as I go.

**Comment by ray-kast** [comment]
Pushed another commit to master adding group panels that can hold a split.  Header control customization is next on the docket.  As previously mentioned, https://go.ryan-s.net/apm/ has the current trunk build running.

I didn't forget about this issue, I just got hit with a lot of work over the past months.  Hopefully I have enough time now to come back and knock this out!

**Comment by ray-kast** [comment]
> Hopefully I have enough time now to come back and knock this out!
That aged poorly.  Anyway, I'm working on handling rendering and configuration.  Got a primitive form of custom header rendering done, but tabs and tab rows don't work yet since they function differently.  After that it's just a matter of performance fixes and configurability.
